parser-typechecker/src/Unison/Codebase/Editor/HandleCommand.hs

```diff
@@ -115,8 +117,8 @@ commandLine config awaitInput setBranchRef rt notifyUser notifyNumbered loadSour
     Typecheck ambient names sourceName source -> do
       -- todo: if guids are being shown to users,
       -- not ideal to generate new guid every time
-      namegen <- Parser.uniqueBase58Namegen
-      let env = Parser.ParsingEnv namegen names
+      let namegen = Parser.uniqueBase32Namegen rng
+          env = Parser.ParsingEnv namegen names
```

Comment:
I just spotted something else that I misread last night but is quite important. rng should be an ‘IO gen’ here and you will need to do a bind to obtain the rng. We want a fresh random number generator each time we typecheck. Otherwise the generator will produce the same values - notice it’s a pure interface.

For main you’ll pass in the IO action that grabs the system rng. It’ll be different every time since it’s seeded from /dev/urandom or something.

For the transcript runner, it’s not quite as simple as ‘pure constRng’, since you want to update the state of the rng after each use. I think something like - keep an IORef with an rng in it, and produce an IO action that reads that rng, and returns it but also sets the IORef with an updated version of the rng.

Lmk if that makes any sense.

Reply:
This should be fixed now, we now take a function that takes an integer and returns an `IO Gen`. We pass it a different integer every time (incrementing by 1). It's ignored when called from main and it just returns a new std rng. We use the number as the rng seed when calling from the transcript runner.